src/search/search_engines/search_common.cc

```diff
@@ -80,16 +80,28 @@ shared_ptr<OpenListFactory> create_greedy_open_list_factory(
   If w = 0, we omit the h-evaluator altogether:
   we use g instead of g + 0 * h.
 */
-static shared_ptr<Evaluator> create_wastar_eval(const shared_ptr<GEval> &g_eval, int w,
+static shared_ptr<Evaluator> create_wastar_eval(const Options &options,
+                                                const shared_ptr<GEval> &g_eval, int w,
                                                 const shared_ptr<Evaluator> &h_eval) {
     if (w == 0)
         return g_eval;
     shared_ptr<Evaluator> w_h_eval = nullptr;
-    if (w == 1)
+    if (w == 1) {
         w_h_eval = h_eval;
-    else
-        w_h_eval = make_shared<WeightedEval>(h_eval, w);
-    return make_shared<SumEval>(vector<shared_ptr<Evaluator>>({g_eval, w_h_eval}));
+    } else {
```

Comment:
This entire `else` case seems a bit tedious, I definitely preferred the way it was before for this particular piece of code. Unfortunately, I don't see a better solution for now...

Reply:
Objects which are created from the command line should not be created from code, in principle. But getting rid of these things is only possible after solving the component interaction problem.

Reply:
We had similar discussions in the past. There are several drawbacks to using `Options` objects in this way, which is why we had these other "direct" constructors. These are some of the main points again `Options`:

1. Requiring `Options` loses static type safety (i.e. checking at compile-time that we have the right names and types of parameters). Some things are not even checked dynamically. For example, if we add an option that doesn't exist, we won't notice.
2. Requiring `Options` introduces unnecessary overhead due to the hash-map operations and type conversions to and from `Any` objects involved.
3. Everything now has a dependency on `Options`. Even people that only want to use a fixed planner configuration for which they could in principle hard-code the constructor have to import significant parts of our option-parsing code. This can be viewed as algorithmic code depending on user-interface code, which is generally bad. (Although it's a matter of opinion whether `Options` should could as user-interface code.)
4. The code that needs to construct `Options` manually for things like here is tedious.
 
Some of these can be addressed by having two constructors like we did in these cases in the past, one taking its arguments directly and one taking `Options`. It's not something we wanted to do in all cases because of the boiler-plate involved. Also, the non-Options-based constructor calls can be hard to read when there are many arguments due to C++'s lack of support for something like keyword arguments.

It is worth reconsidering these arguments after substantial changes to the parts of the planner that influenced our decision at the time. For example, perhaps this is something we should rediscuss once we introduce builders. Perhaps builders should use Options objects, and the objects they build should not. If you think this might be the case, perhaps we can save parts of this comment and its context in a relevant place of the issue tracker.

Reply:
I forgot to mention: it might very well be possible to address 4. by working on the interface of Options objects. In principle, I think it might be possible to support something like

`Options options {
  {"w", 3},
  {"evaluator", eval},
  {"use_foobar", true} };`

although care would need to be taken that C++ derives the correct types.

Reply:
I agree that using Options is bad. To circumvent this problem here, we would also need to have a second constructor in the evaluator base class that takes the LogProxy instead of the Options object. Is this the preferred solution?

Reply:
I think it's OK to use `Options` here, I don't think it's the only place like this. But we should look at it again once we implement the builders.